#15 Cucumber test with undefined steps are passing, Upgrading to cucu…

mat013 opened the pull request:
…mber-reporting 2.5.0, which provides properties to check for undefined steps,

Signed-off-by: Meang Akira Tanaka <meang.tnk@gmail.com>

mat013 commented:
Hi Jay 

I have had trouble running the integration test on my machine so I omitted that task, which I am being punished by :p.

For some reason I can't access the cucumber gradle plugin when I run the test at home. because it is not in the repository. Do you need to do something in order to get access to the plugin 

br

Meang

JayStGelais commented:
Meang

I realized that I forgot to add the changes to the buildScript block to the sample configuration in the README. Try what I have there now.

mat013 commented:
Hi Jay

I think I have several issues, but let us start with the first one.
I am running windows 10. It shouldn't mean anything but just so you know.

1. when I run gradlew build

integration test failed and it wrote that I should look at following html file:

file:///E:/opensource/gradle-cucumber-jvm-plugin/build/reports/integTest/classes/com.commercehub.gradle.cucumber.CucumberIntegrationSpec.html

I looked into that and I could see the following issues in the attached html file(which I renamed to txt as github does not allowe html)

[Test results - Class com.commercehub.gradle.cucumber.CucumberIntegrationSpec.txt](https://github.com/commercehub-oss/gradle-cucumber-jvm-plugin/files/382359/Test.results.-.Class.com.commercehub.gradle.cucumber.CucumberIntegrationSpec.txt)

I investigated it further and could see that in happy path a script is being created and it does not have any buildScript block.

So my question is shouldn't it have one?

I added a build block in the CucumberIntegrationSpec.groovy, so they were generated and another issue happened and that was 
[Test results - Class com.commercehub.gradle.cucumber.CucumberIntegrationSpec2.txt](https://github.com/commercehub-oss/gradle-cucumber-jvm-plugin/files/382366/Test.results.-.Class.com.commercehub.gradle.cucumber.CucumberIntegrationSpec2.txt)

the file features-happypath-feature.html however a similar file with a _ at the end of the filename exist i.e. features-happypath-feature_.html

I can easily fix the two issues two work on my machine but it does not seems right... :)

Do you have any suggestion what the reason can be why it does not work on my machine. Could it be that you have in your personal gradle settings have a accesses your repository?

I am unsure if the second issue is because I upgraded the masterthought framework... but let us see when the first issue has been solved

mat013 commented:
I can also see the the CI is failing, how do I get access to the file it is refering to... or alternative can you send it to me?

JayStGelais commented:
The integration tests do not require the buildScript block because they get the plugin from the local build. I do not know why you are getting the '_' after feature, but a change based on mastersthoughts upgrade could e the culprit. If that is indeed how the new version of the report library will name the file, it would be OK to change the assertion.

JayStGelais commented:
Integrations tests are still failing. I pulled your changes locally and see that the file being generated on my system is named:

`build/reports/test/features-happypath-feature_feature.html`

Is it really different on your system (Windows, right?). If so then we should probably make that test assertion more flexible. The file can be named anything so long as:

1. It references the feature (has string 'happypath' in it)
2. Is linked properly from the feature summary, which it is.

Why don'
t you update the test to compare with a regex. something like 'features-.*happypath.*.html'. Then push again once tests are passing.

mat013 commented:
Ahhh at last... The file on windows seems to differ from whatever operating system you are running.

JayStGelais commented on src/main/groovy/com/commercehub/gradle/cucumber/CucumberRunner.groovy:
```diff
@@ -47,31 +48,13 @@ class CucumberRunner {
-                }
-                args << '--snippets'
-                args << options.snippets
+                applyGlueArguments(args)
```
Big fan of this refactor!

mat013 replied:
Thanks. 

I kept bumping into the ABC metric in codenarc...
